framework/src/main/java/org/checkerframework/framework/type/DefaultTypeHierarchy.java

```diff
@@ -285,104 +283,101 @@ protected boolean areEqualInHierarchy(
   }
 
   /**
-   * A declared type is considered a supertype of another declared type only if all of the type
-   * arguments of the declared type "contain" the corresponding type arguments of the subtype.
-   * Containment is described in <a
+   * Returns true if {@code outside} contains {@code inside}, that is, if the set of types denoted
+   * by {@code outside} is a superset of or equal to the set of types denoted by {@code inside}.
+   *
+   * <p>Containment is described in <a
    * href="https://docs.oracle.com/javase/specs/jls/se11/html/jls-4.html#jls-4.5.1">JLS section
-   * 4.5.1 "Type Arguments of Parameterized Types"</a>.
+   * 4.5.1 "Type Arguments of Parameterized Types"</a>. A declared type is considered a supertype of
+   * another declared type only if all of the type arguments of the declared type "contain" the
+   * corresponding type arguments of the subtype.
    *
-   * @param inside a type argument of the "subtype"
-   * @param outside a type argument of the "supertype"
+   * @param inside a possibly-contained type
+   * @param outside a possibly-containing type
    * @param canBeCovariant whether or not type arguments are allowed to be covariant
    * @return true if inside is contained by outside, or if canBeCovariant == true and {@code inside
    *     <: outside}
    */
   protected boolean isContainedBy(
-      final AnnotatedTypeMirror inside, final AnnotatedTypeMirror outside, boolean canBeCovariant) {
+      AnnotatedTypeMirror inside, AnnotatedTypeMirror outside, boolean canBeCovariant) {
 
     if (ignoreUninferredTypeArgument(inside) || ignoreUninferredTypeArgument(outside)) {
```

Comment:
I find the name of this method a bit hard to read; it sounds more like an imperative or command than a predicate.  Consider changing to "shouldIgnore..." or "isIgnored..." or the like.

Also, the Javadoc of this method says that it returns true if the checker should not issue warnings, but here it seems to affect behavior too.

Reply:
I renamed the method. 

> Also, the Javadoc of this method says that it returns true if the checker should not issue warnings, but here it seems to affect behavior too.

I don't understand this statement.  It does return true if the checker should not issue warnings.  In order for the checker to not issue warnings, `isSubtype` returns true.

Reply:
In the common case, a method either return a value or has a side effect, but not both.  I think this one does both.  Given that it does both, I think it should be documented as such, but right now it is only documented as returning a value and the side effect is not mentioned in the documenation.